### ISSUE TITLE
[F] Add soloc option to add_model call

### ISSUE THREAD

**Issue by kylebaron** [dpastoor/overseer#5]
```sh
.modelcache/
├── code.cpp
└── mrgsolve-so-0.7.9.9002-x86_64-apple-darwin13.4.0
    └── code
        ├── code-mread-source.cpp
        ├── code-mread-source.o
        ├── code-mread-source.so
        ├── code-so-109b31f72497a.so
        └── mrgmod_cache.RDS

```

**Comment by dpastoor** [comment]
move discussion to here rather than closed pull request:

out of curiousity, where would you use `add_model()`?

eg, are you using it in something like models.R, or are you using it to allow overseer to bring in a set of models, then in the file itself add another model just to have it in overseer?

**Comment by dpastoor** [comment]
there is somewhat of an inherent tension I'm finding between the behavior of `add_model()` and `add_model_file()` then `use()`, since with add_model() when you source you now have to re-source the entire overseer object to update said model (given tracking it in the models.R), vs with add_model_file() it is only keeping track of the file itself then loading on use(), so it allows you to keep tweaking and just re-`use()`ing to re-compile it

**Comment by kylebaron** [comment]
In the `models.R` file.  Is that the way we should be bringing in an inline model?

**Comment by dpastoor** [comment]
sure - absolutely works, but as per my comment above, it feels ungainly, as you lose all control once it's sourced in, vs add_model_file feels (to me) like a superior option since its much easier to dynamically source.

I'm not going to get rid of it or anything crazy, just trying to figure out if it will illicit too much confusion, and how to mitigate that.

**Comment by kylebaron** [comment]
Ah ... I wasn't that sophisticated with it yet.  I don't have any active project work right now, so I'm still using this in a pretty simple setting.  Also, I tend to run everything from the top `option-command-b` so I usually end up re-sourcing everything anyway.  Obviously, on a larger script, maybe won't be doing that as much.

**Comment by dpastoor** [comment]
gotcha - makes sense - and I think that's a reasonable way of using it. I think I'll just have to be very clear in the documentation. The other thing that I like about the separate files, is down the line it would be easier to track the models in an external program that could do 'things' - eg a shiny app running a separate process, watching all files for changes, and on file change, try to recompile and return any errors.

Then of course the obvious nature of it just being a nice separation of concerns for tracking via git etc.

**Comment by kylebaron** [comment]
I totally agree on preference for "separate files" approach.  We end up doing a lot of inline models when showing examples or in the course.  But I always implore people to use a separate file when really doing production work.  Seems like clear documentation should be enough to warn the users.
